src/library/export/trackexport.cpp

```diff
@@ -0,0 +1,129 @@
+            return;
+        }
+        ++i;
+        emit(progress(fileinfo.fileName(), i, m_filenames.size()));
+    }
+}
+
+void TrackExport::exportTrack(QString sourceFilename) {
+    QFileInfo source_fileinfo(sourceFilename);
+    const QString dest_filename = m_destDir + "/" + source_fileinfo.fileName();
+    QFileInfo dest_fileinfo(dest_filename);
+
+    if (dest_fileinfo.exists()) {
+        switch (m_overwriteMode) {
+        // Give the user the option to overwrite existing files in the destination.
+        case OverwriteMode::ASK:
+            switch (makeOverwriteRequest(dest_filename)) {
+            case OverwriteAnswer::SKIP:
+            case OverwriteAnswer::SKIP_ALL:
+                qDebug() << "skipping" << sourceFilename;
+                return;
+            case OverwriteAnswer::OVERWRITE:
+            case OverwriteAnswer::OVERWRITE_ALL:
+                break;
+            case OverwriteAnswer::CANCEL:
+                m_errorMessage = tr("Export process was canceled");
+                stop();
+                return;
+            }
+            break;
+        case OverwriteMode::SKIP_ALL:
+            qDebug() << "skipping" << sourceFilename;
+            return;
+        case OverwriteMode::OVERWRITE_ALL:;
+        }
+
+        // Remove the existing file in preparation for overwriting.
+        QFile dest_file(dest_filename);
+        qDebug() << "Removing existing file" << dest_filename;
+        if (!dest_file.remove()) {
+            const QString error_message =
+                tr("Error removing file %1: %2. Stopping.").arg(
+                dest_filename, dest_file.errorString());
+            qWarning() << error_message;
+            m_errorMessage = error_message;
+            m_bStop = true;
+            return;
+        }
+    }
+
+    qDebug() << "Copying" << sourceFilename << "to" << dest_filename;
+    QFile source_file(sourceFilename);
+    if (!source_file.copy(dest_filename)) {
+        const QString error_message =
+                tr("Error exporting track %1 to %2: %3. Stopping.").arg(
+                sourceFilename, dest_filename, source_file.errorString());
+        qWarning() << error_message;
+        m_errorMessage = error_message;
+        m_bStop = true;
+        return;
+    }
+}
+
+TrackExport::OverwriteAnswer TrackExport::makeOverwriteRequest(QString filename) {
+    // QT's QFuture is not quite right for this type of threaded question-and-answer.
+    // std::future works fine, even with signals and slots.
+    QScopedPointer<std::promise<OverwriteAnswer>>
+            mode_promise(new std::promise<OverwriteAnswer>());
+    std::future<OverwriteAnswer> mode_future = mode_promise->get_future();
+
+    emit(askOverwriteMode(filename, mode_promise.data()));
+
+    // Block until the user tells us the answer, but check for cancelation
+    // as well.
+    std::future_status status(std::future_status::timeout);
+    while (status != std::future_status::ready && !m_bStop) {
```

Comment:
OK 500 ms is not CPU burning. But it would be nice if we can avoid such polling loops. 
What is the case that  changes m_stop during this loop?
If required, there are concepts available to wait for one of two or more conditions.